imap/http_carddav.c

```diff
@@ -622,6 +615,15 @@ static int carddav_store_resource(struct transaction_t *txn,
         }
     }
 
+    /* Check size of vCard (allow existing oversized cards to be updated) */
+    struct buf *buf = vcard_as_buf(vcard);
+    if ((buf_len(buf) > (size_t) vcard_max_size) &&
+        !(oldrecord && (oldrecord->size - oldrecord->header_size) > (size_t) vcard_max_size)) {
```

Comment:
I had to wrap my brain around the negated condition. I think it would be simpler to read without the negation, such as ` .. && (!oldrecord || (oldrecord->size - oldrecord->header_size) <= (size_t) vcard_max_size))`. That's also then symmetric with how it is written in carddav_db.c

Reply:
Fixed